tests/unittest/llmapi/test_llm_args.py

```
@@ -381,6 +381,38 @@ def test_SleepConfig_restore_modes_normalized_from_defaultdict():
+    """SleepConfig must survive a pickle round-trip.
+
+    MPI worker initialisation serialises llm_args (including SleepConfig) via
+    pickle to distribute configuration to each rank.  The defaultdict inside
+    restore_modes previously used a closure lambda as its default_factory, which
+    is not picklable.  This test catches any regression to that pattern.
+    """
+    import pickle
+
+    # Default construction
+    cfg_default = SleepConfig()
+    rt = pickle.loads(pickle.dumps(cfg_default))
```

Ruff (0.15.12)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)